src/Compilers/Core/Portable/InternalUtilities/SemaphoreSlimExtensions.cs

```diff
@@ -14,13 +14,14 @@ public static SemaphoreDisposer DisposableWait(this SemaphoreSlim semaphore, Can
             return new SemaphoreDisposer(semaphore);
         }
 
-        public async static Task<SemaphoreDisposer> DisposableWaitAsync(this SemaphoreSlim semaphore, CancellationToken cancellationToken = default)
+        public static async ValueTask<SemaphoreDisposer> DisposableWaitAsync(this SemaphoreSlim semaphore, CancellationToken cancellationToken = default)
```

Comment:
📝 This change was merged in #36128, largely due to the groundwork laid by this pull request